### PR TITLE
Added kitchen-dokken testing for faster feedback with a lighter footprint

### PR DESCRIPTION
**Pull request by espoelstra**
Currently it is running two containers so slightly heavier than just testing with 18.04, but it ensures pre-systemd and systemd compatibility.

Only requirements in this case is Docker and the ChefDK, no need for Vagrant + Virtualbox and a bunch of plugins for Vagrant.

### CODE REVIEW

File: jenkins_wrapper_cookbook/nodes/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This file exists to allow checking in the empty directory required by Chef-zero
```
Comment (espoelstra) [inline jenkins_wrapper_cookbook/nodes/.gitignore:1]: This is actually to fix the Vagrant chef-zero provisioner, as it requires a `nodes_path` and fails to start if the directory doesn't exist.

File: jenkins_wrapper_cookbook/recipes/default.rb [hunk 2]
```diff
@@ -86,7 +97,7 @@
 
 execute 'install_plugins' do
   command <<-EOH.gsub(/^ {4}/,'')
-  source /etc/profile
+  . /etc/profile
```
Comment (espoelstra) [inline jenkins_wrapper_cookbook/recipes/default.rb:101]: Because the shell interpreter isn't defined it defaults to `/bin/sh` which on Ubuntu/Debian is `/bin/dash` which doesn't support the Bash-ism `source`. The `.` version of `source` works on any POSIX system.

File: jenkins_wrapper_cookbook/Vagrantfile [hunk 3]
```diff
@@ -47,6 +47,18 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     chef.data_bags_path = 'data_bags'
     chef.environments_path = 'environments'
     chef.nodes_path = 'nodes'
+    # Need to clean 'nodes' before each provision otherwise it uses existing data
+    if File.exist?(chef.nodes_path)
+      if File.directory?(chef.nodes_path)
+          Dir.foreach(chef.nodes_path) do |file|
+            if ((file.to_s != ".")\
+              and (file.to_s != "..")\
+              and (file.to_s != ".gitignore"))
```
Comment (espoelstra) [inline jenkins_wrapper_cookbook/Vagrantfile:56]: You could add additional files in here if you were actually testing with node data, though I haven't tested this myself because the chef-zero provisioner typically generates the data itself.
